packages/turf-nearest-point-on-line/index.ts

```diff
@@ -52,96 +53,111 @@ function nearestPointOnLine<G extends LineString | MultiLineString>(
 
   let closestPt: Feature<
     Point,
-    { dist: number; index: number; location: number }
+    { dist: number; index: number; multiFeatureIndex: number; location: number }
   > = point([Infinity, Infinity], {
     dist: Infinity,
     index: -1,
+    multiFeatureIndex: -1,
     location: -1,
   });
 
   let length = 0.0;
-  flattenEach(lines, function (line: any) {
-    const coords: any = getCoords(line);
+  flattenEach(
+    lines,
+    function (line: any, _featureIndex: number, multiFeatureIndex: number) {
```

Comment:
should we add the `featureIndex` too since we have it?

Reply:
The method is having an identify crisis, it's called `nearestPointOn**Line**` (singular) implying it only accepts one line, however the first argument is named **`lines`** (plural) implying it accepts multiple lines, the lines argument type is defined to accept `Geometry|Feature<LineString|MultiLineString>` which implies only a single line `Feature` is accepted, albeit you can have a `MultiLineString` geometry within that single `Feature`, however it may also accept multiple lines through a `GeometryCollection` type of `Geometry`.

Checking the implementation it just passes `lines` through to `flattenEach` which accepts `FeatureCollection | Feature | Geometry` thereby giving rise to the `featureIndex`.

We should decide that either:

1. the function is to only accept a single line feature and therefore doesn't need to deal with `featureIndex` or,
2. the function can accept multiple features and therefore also include the `featureIndex` in the result.

While officially supporting `FeatureCollection` sounds like a good idea, it gets complicated when that `FeatureCollection` contains multiple geometry types, and defining how the method behaves in those situations is messy. On the other hand we already support `GeometryCollection` which has the same issue. :shrug:

I think for now we go ahead and just include the `multiFeatureIndex` and leave the decision on `featureIndex` to another discussion/issue/PR.

Reply:
Fair points, thanks for the detailed writeup!